### DISCUSSION TITLE
why homebrew distributes llvm with all its subprojects?

### DISCUSSION THREAD

**Discussion by paperchalice** [Homebrew/discussions#157] [Everyday usage] [answered]
LLVM has so many subprojects, some subprojects have already the formula in homebrew e.g. libomp, clang-format. Some projects may be installed improper as keg-only, e.g. flang, lld. So why homebrew choose this way to distribute llvm?

**Comment by SMillerDev** [comment, marked as answer]
> some subprojects have already the formula in homebrew e.g. libomp, clang-format

Some formula are added because people want to just have a single subproject to use with the Xcode LLVM.

> Some projects may be installed improper as keg-only, e.g. flang, lld.

This happens when something conflicts or conflicted with a system version of a dependency.
